src/mptool.f90

```diff
@@ -0,0 +1,331 @@
+!Copyright (c) 2012-2022, Xcompact3d
+!This file is part of Xcompact3d (xcompact3d.com)
+!SPDX-License-Identifier: BSD 3-Clause
+module mptool
+  !
+  use decomp_2d_constants, only : mytype, real_type
+  use decomp_2d_mpi, only: nrank, nproc
+  !
+  implicit none
+  !
+  interface psum
+    module procedure psum_mytype_ary
+    module procedure psum_integer
+    module procedure psum_mytype
+  end interface
+  !
+  interface pmax
+    module procedure pmax_int
+    module procedure pmax_mytype
+  end interface
+  !
+  interface ptabupd
+    module procedure ptable_update_int_arr
+    module procedure updatable_int
+  end interface ptabupd
+  !
+  !
+  interface pgather
+    module procedure pgather_int
+  end interface
+  !
+  ! integer(hid_t) :: h5file_id
+  !
+  integer :: mpi_comm_particle,mpi_rank_part,mpi_size_part
+  !
+  character(len=4) :: rankname
+  !
+  contains
+  !
+  !+-------------------------------------------------------------------+
+  !| This subroutine is used to finalise mpi and stop the program.     |
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 19-July-2019: Created by J. Fang @ STFC Daresbury Laboratory      |
+  !+-------------------------------------------------------------------+
+  subroutine mpistop
+    !
+    use mpi
+    !
+    integer :: ierr
+    !
+    call mpi_barrier(mpi_comm_world,ierr)
+    !
+    call mpi_finalize(ierr)
```

Comment:
This is called at the end of the subroutine `finalise_xcompact3d` already

Reply:
In case of error, one can use `decomp_2d_abort`

https://github.com/2decomp-fft/2decomp-fft/blob/main/src/decomp_2d_mpi.f90#L23

Reply:
We have removed all unnecessary code